test/e2e/storage/vsphere_statefulsets.go

```diff
@@ -0,0 +1,141 @@
+	storageclassname = "nginx-sc"
+)
+
+var _ = SIGDescribe("vsphere statefulset", func() {
+	f := framework.NewDefaultFramework("vsphere-statefulset")
+	var (
+		namespace string
+		client    clientset.Interface
+	)
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		namespace = f.Namespace.Name
+		client = f.ClientSet
+	})
+	AfterEach(func() {
+		framework.Logf("Deleting all statefulset in namespace: %v", namespace)
+		framework.DeleteAllStatefulSets(client, namespace)
+	})
+
+	It("vsphere statefulset testing", func() {
+		By("Creating StorageClass for Statefulset")
+		scParameters := make(map[string]string)
+		scParameters["diskformat"] = "thin"
+		scSpec := getVSphereStorageClassSpec(storageclassname, scParameters)
+		sc, err := client.StorageV1().StorageClasses().Create(scSpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(sc.Name, nil)
+
+		By("Creating statefulset")
+		statefulsetTester := framework.NewStatefulSetTester(client)
+		statefulset := statefulsetTester.CreateStatefulSet(manifestPath, namespace)
+		replicas := *(statefulset.Spec.Replicas)
+		// Waiting for pods status to be Ready
+		ssPods := statefulsetTester.GetPodList(statefulset)
+		Expect(ssPods.Items).NotTo(BeEmpty(), fmt.Sprintf("Unable to get list of Pods from the Statefulset: %v", statefulset.Name))
+		Expect(len(ssPods.Items) == int(replicas)).To(BeTrue(), "Number of Pods in the statefulset should match with number of replicas")
+		statefulsetTester.WaitForStatusReadyReplicas(statefulset, replicas)
+		Expect(statefulsetTester.CheckMount(statefulset, mountPath)).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("Scaling down statefulsets to number of Replica: %v", replicas-1))
+		_, scaledownErr := statefulsetTester.Scale(statefulset, replicas-1)
+		Expect(scaledownErr).NotTo(HaveOccurred())
+		statefulsetTester.WaitForStatusReadyReplicas(statefulset, replicas-1)
+
+		vsp, err := vsphere.GetVSphere()
+		Expect(err).NotTo(HaveOccurred())
+
+		// After scale down, verify vsphere volumes are detached from deleted pods
+		By("Verify Volumes are detached from Nodes after Statefulsets is scaled down")
+		for _, sspod := range ssPods.Items {
+			_, err := client.CoreV1().Pods(namespace).Get(sspod.Name, metav1.GetOptions{})
+			if err != nil {
+				Expect(apierrs.IsNotFound(err), BeTrue())
+				for _, volumespec := range sspod.Spec.Volumes {
+					if volumespec.PersistentVolumeClaim != nil {
+						vSpherediskPath := getvSphereVolumePathFromClaim(client, statefulset.Namespace, volumespec.PersistentVolumeClaim.ClaimName)
+						framework.Logf("Waiting for Volume: %q to detach from Node: %q", vSpherediskPath, sspod.Spec.NodeName)
+						Expect(waitForVSphereDiskToDetach(vsp, vSpherediskPath, types.NodeName(sspod.Spec.NodeName))).NotTo(HaveOccurred())
+					}
+				}
+			}
+		}
+
+		By(fmt.Sprintf("Scaling up statefulsets to number of Replica: %v", replicas+1))
+		_, scaleupErr := statefulsetTester.Scale(statefulset, replicas+1)
```

Comment:
nit: create `newReplicas =  rep[licas+1` and use that below.

Reply:
I think it's fine, `replicas-1` was used above too.